module/zfs/zfs_ioctl.c

```diff
@@ -3737,7 +3738,9 @@ zfs_check_settable(const char *dsname, nvpair_t *pair, cred_t *cr)
 			    SPA_VERSION_ZLE_COMPRESSION))
 				return (SET_ERROR(ENOTSUP));
 
-			if (intval == ZIO_COMPRESS_LZ4) {
+			if (intval == ZIO_COMPRESS_LZ4 ||
+			    (intval >= ZIO_COMPRESS_LZ4HC_1 &&
```

Comment:
That is wrong indeed (the code in this pull request). Code in lz4hc-hack avoids having to make a new feature ([[1]](https://github.com/vozhyk-/zfs/commit/0da8e908132f166bb2c75465eac4342c0d91de5a), [[2]](https://github.com/vozhyk-/zfs/commit/9e3da749baeab6a432672dd8f72bda5821318568)) (but it's still a hack).